runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -480,7 +492,11 @@ function LanguageTree:_async_parse(range, on_parse)
   local ct = buf.changedtick
   local total_parse_time = 0
   local redrawtime = vim.o.redrawtime
-  local timeout = not vim.g._ts_force_sync_parsing and default_parse_timeout_ms or nil
+
+  local thread_state = {} ---@type ParserThreadState
```

Comment:
Why does this need to be a table? If it should be a table, I think it should also include `ct`, because it's also a mutable state used in `ct`.

Reply:
I just thought it would look more organized if all the mutable thread states are collected into a single table (including `total_parse_time`, which I missed in the previous comment). But in terms of functionality there is no difference. Please feel free to disregard.